service/systemd/service.go

```diff
@@ -87,6 +89,7 @@ func NewService(name string, conf common.Conf, dataDir string, newDBus DBusAPIFa
 		ConfName: confName,
 		UnitName: confName,
 		DirName:  dirName,
+		OldDir:   "/var/lib/juju/init",
```

Comment:
Rather than hard-coding the value here, it should be passed in from `system.newService`, where `/lib/systemd/system` is passed in. You'll need a new parameter for systemd.NewService.

Since everything else is xName, I think it would be better called `OldDirName` or `FallbackDirName`.